core/src/databases/table.rs

```diff
@@ -170,10 +171,12 @@ impl Table {
         self.schema = Some(schema);
     }
 
+    // if search_store is provided, delete the table node from the search index
     pub async fn delete(
         &self,
         store: Box<dyn Store + Sync + Send>,
         databases_store: Box<dyn DatabasesStore + Sync + Send>,
+        search_store: Option<Box<dyn SearchStore + Sync + Send>>,
```

Comment:
why optional?

Reply:
When deleting a datasource, there's a batch delete on all nodes querying on datasource
I thought in that case it was costly and useless to delete tables one by one (ES average delete is likely ~5-10ms)